Source/Scene/ModelExperimental/SceneMode2DPipelineStage.js

```diff
@@ -0,0 +1,284 @@
+
+function computeBoundingSphere2D(renderResources, modelMatrix, frameState) {
+  // Compute the bounding sphere in 2D.
+  const transformedPositionMin = Matrix4.multiplyByPoint(
+    modelMatrix,
+    renderResources.positionMin,
+    scratchProjectedMin
+  );
+
+  const projectedMin = SceneTransforms.computeActualWgs84Position(
+    frameState,
+    transformedPositionMin,
+    transformedPositionMin
+  );
+
+  const transformedPositionMax = Matrix4.multiplyByPoint(
+    modelMatrix,
+    renderResources.positionMax,
+    scratchProjectedMax
+  );
+
+  const projectedMax = SceneTransforms.computeActualWgs84Position(
+    frameState,
+    transformedPositionMax,
+    transformedPositionMax
+  );
+
+  return BoundingSphere.fromCornerPoints(
+    projectedMin,
+    projectedMax,
+    new BoundingSphere()
+  );
+}
+
+const scratchPosition = new Cartesian3();
+
+function dequantizePositionsTypedArray(typedArray, quantization) {
+  // Draco compression is normally handled in the dequantization stage
+  // in the shader, but it must be decoded here in order to project
+  // the positions to 2D / CV.
+  const length = typedArray.length;
+  const dequantizedArray = new Float32Array(length);
+  const quantizedVolumeOffset = quantization.quantizedVolumeOffset;
+  const quantizedVolumeStepSize = quantization.quantizedVolumeStepSize;
+  for (let i = 0; i < length; i += 3) {
+    const initialPosition = Cartesian3.fromArray(
+      typedArray,
+      i,
+      scratchPosition
+    );
+    const scaledPosition = Cartesian3.multiplyComponents(
+      initialPosition,
+      quantizedVolumeStepSize,
+      initialPosition
+    );
+    const dequantizedPosition = Cartesian3.add(
+      scaledPosition,
+      quantizedVolumeOffset,
+      scaledPosition
+    );
+
+    dequantizedArray[i] = dequantizedPosition.x;
+    dequantizedArray[i + 1] = dequantizedPosition.y;
+    dequantizedArray[i + 2] = dequantizedPosition.z;
+  }
+
+  return dequantizedArray;
+}
+
+function createPositionsTypedArrayFor2D(
+  attribute,
+  modelMatrix,
+  referencePoint,
+  frameState
+) {
+  const typedArray = attribute.typedArray.slice();
+
+  let result;
+  if (defined(attribute.quantization)) {
+    // Dequantize the positions if necessary.
+    result = dequantizePositionsTypedArray(
+      attribute.typedArray,
+      attribute.quantization
+    );
+  } else {
+    result = new Float32Array(
+      typedArray.buffer,
+      typedArray.byteOffset,
+      typedArray.byteLength / Float32Array.BYTES_PER_ELEMENT
+    );
+  }
+
+  const startIndex = attribute.byteOffset / Float32Array.BYTES_PER_ELEMENT;
+  const length = result.length;
+  for (let i = startIndex; i < length; i += 3) {
+    const initialPosition = Cartesian3.fromArray(result, i, scratchPosition);
+
+    const transformedPosition = Matrix4.multiplyByPoint(
+      modelMatrix,
+      initialPosition,
+      initialPosition
+    );
+
+    const projectedPosition = SceneTransforms.computeActualWgs84Position(
```

Comment:
I was trying the [San Francisco Ferry Building Sandcastle](http://localhost:8080/Apps/Sandcastle/index.html?src=3D%20Tiles%20Next%20Photogrammetry%20Classification.html&label=3D%20Tiles%20Next) and this line seems to hit a divide by zero error when I zoom in to the tileset.

![image](https://user-images.githubusercontent.com/8422414/170132084-d8e44b6b-6212-480c-8093-d50293a71495.png)